proto/lighthouse-result.proto

```diff
@@ -85,6 +85,9 @@ message LighthouseResult {
 
     // The benchmark index that indicates rough device class
     google.protobuf.DoubleValue benchmark_index = 3;
+
+    // The version of libraries with which these results were generated. Ex: axe-core.
+    map<string, string> credits = 4;
```

Comment:
not sure I'm qualified to judge the appropriateness of different protobuf structures :)

cc @exterkamp to weigh in there?

Reply:
This is fine if the keys are relatively dynamic, or if there are lots (like audit id's).

Otherwise we could make an inner message that would keep this more defined.

```
message Credits {
  string axe_core = 1;
}

Credits credits = 4;
```

It seems like this is well known and that there won't be very many? So I almost want to err on the side of an inner message to keep things strict.

Reply:
Would a proto consumer be able to iterate that explicitly defined structure? because that's probably the only reasonable use case of this (that's not what we're doing in the report yet, but if this expands to more than `axe-core` I see us doing a generic loop)

Reply:
Why would someone want to just loop generically? That doesn't seem like something someone would do? It seems like a (power-)user would know what key they want b/c they want to look for a specific version of something. Like, I could see this being a use:

```
for key, val in credits {
  if key == "axe-core" {
    if val != 2.1 {
      return error(bad)
    }
  }
  // more checks on keys
}
```
imo it is worse DX than:
```
if credits.getAxeCore() != 2.1 {
  return error(bad)
}
// more checks on other Credits fields
```

Also, this kind of nested proto structure doesn't lend itself _well_ to iteration. You can reflectively inspect the message and iterate on the fields in a loop, but that's not like "great".

Reply:
It has trouble transforming `axe-core` to the proto `axe_core`.

![image](https://user-images.githubusercontent.com/4071474/91346490-8cbceb00-e7a6-11ea-85db-c550d2be4613.png)

Reply:
Talked a bit more about use cases and having the key string match exactly with the npm name, e.g. "axe-core" is desirable. So the map<string, string> LGTM.